pylintrc

```diff
@@ -239,7 +239,7 @@ ignore-docstrings=yes
 ignore-imports=no
 
 # Minimum lines number of a similarity.
-min-similarity-lines=4
+min-similarity-lines=8
```

Comment:
It is true, I was trying to be a bit lazy. :)

The problem is that we have code duplication between `CatalogSearchIterator`, `GeopediaFeatureIterator`, and `WebFeatureService` (although the last one uses a bit different names for attributes). Then I didn't want to dive into Geopedia code only for this issue.

The solution for this would be to have a base iteration class, e.g. `FeatureIterator` from which all `3` classes would inherit. Additionally, also `_iter_pages`, which we use for batch and BYOC, could be extended into a proper iterator class (currently it is only a generator function). Then also that could inherit from `FeatureIterator` and it would be also backward compatible with the current implementation of `_iter_pages`.

If you agree I would do the following: 
- open an issue about this,
- once we merge Catalog and BYOC PRs I would prepare another PR that would unify all iterators throughout the package.

Reply:
Absolutely.